There is an error while running the Infinispan tests:

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.quarkus.it.infinispan.client.InfinispanClientFunctionalityInGraalITCase
Feb 16, 2020 2:17:30 AM org.infinispan.registry.impl.InternalCacheRegistryImpl registerInternalCache
WARN: ISPN000569: Unable to persist Infinispan internal caches as no global state enabled
Feb 16, 2020 2:17:30 AM org.infinispan.factories.GlobalComponentRegistry preStart
INFO: ISPN000128: Infinispan version: Infinispan 'Turia' 10.1.2.Final
Feb 16, 2020 2:17:31 AM org.infinispan.lock.impl.ClusteredLockModuleLifecycle cacheManagerStarted
INFO: ISPN029009: Configuration is not clustered, clustered locks are disabled
Feb 16, 2020 2:17:31 AM org.infinispan.transaction.lookup.GenericTransactionManagerLookup useDummyTM
INFO: ISPN000104: Using EmbeddedTransactionManager
Feb 16, 2020 2:17:31 AM org.infinispan.server.hotrod.test.HotRodTestingUtil startHotRodServer
INFO: Start server in port 11232
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.062 s <<< FAILURE! - in io.quarkus.it.infinispan.client.InfinispanClientFunctionalityInGraalITCase
[ERROR] io.quarkus.it.infinispan.client.InfinispanClientFunctionalityInGraalITCase  Time elapsed: 2.061 s  <<< ERROR!
org.junit.platform.commons.JUnitException: Quarkus native image start failed, original cause: java.lang.IllegalStateException: Configuration already registered for the given class loader

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   InfinispanClientFunctionalityInGraalITCase » JUnit Quarkus native image start ...
[INFO] 
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
```

--- a/bom/runtime/pom.xml
+++ b/bom/runtime/pom.xml
@@ -119,7 +119,7 @@
         <testng.version>6.14.2</testng.version>
         <assertj.version>3.15.0</assertj.version>
         <json-smart.version>2.3</json-smart.version>
-        <infinispan.version>10.0.0.Final</infinispan.version>
+        <infinispan.version>10.1.2.Final</infinispan.version>
         <infinispan.protostream.version>4.3.1.Final</infinispan.protostream.version>
         <caffeine.version>2.8.0</caffeine.version>
         <netty.version>4.1.44.Final</netty.version>
@@ -2490,7 +2490,11 @@
                 <artifactId>protostream</artifactId>
                 <version>${infinispan.protostream.version}</version>
             </dependency>
-
+            <dependency>
+                <groupId>org.infinispan</groupId>
+                <artifactId>infinispan-component-annotations</artifactId>
+                <version>${infinispan.version}</version>
+            </dependency>
             <dependency>
                 <groupId>org.reactivestreams</groupId>
                 <artifactId>reactive-streams</artifactId>

--- a/extensions/infinispan-embedded/runtime/pom.xml
+++ b/extensions/infinispan-embedded/runtime/pom.xml
@@ -41,6 +41,10 @@
             </exclusions>
         </dependency>
         <dependency>
+            <groupId>org.infinispan</groupId>
+            <artifactId>infinispan-component-annotations</artifactId>
+        </dependency>
+        <dependency>
             <groupId>org.jboss.narayana.jta</groupId>
             <artifactId>narayana-jta</artifactId>
             <optional>true</optional>

--- a/extensions/infinispan-embedded/runtime/src/main/java/io/quarkus/infinispan/embedded/runtime/InfinispanEmbeddedRuntimeConfig.java
+++ b/extensions/infinispan-embedded/runtime/src/main/java/io/quarkus/infinispan/embedded/runtime/InfinispanEmbeddedRuntimeConfig.java
@@ -13,7 +13,7 @@ import io.quarkus.runtime.annotations.ConfigRoot;
 public class InfinispanEmbeddedRuntimeConfig {
 
     /**
-     * The configured Infinispan embeddex xml file which is used by the managed EmbeddedCacheManager and its Caches
+     * The configured Infinispan embedded xml file which is used by the managed EmbeddedCacheManager and its Caches
      */
     @ConfigItem
     public Optional<String> xmlConfig;

--- a/extensions/infinispan-embedded/runtime/src/main/java/io/quarkus/infinispan/embedded/runtime/graal/FixJMXClasses.java
+++ b/extensions/infinispan-embedded/runtime/src/main/java/io/quarkus/infinispan/embedded/runtime/graal/FixJMXClasses.java
@@ -2,6 +2,10 @@ package io.quarkus.infinispan.embedded.runtime.graal;
 
 import java.util.Properties;
 
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+import org.infinispan.configuration.global.GlobalConfiguration;
 import org.infinispan.configuration.global.GlobalJmxStatisticsConfiguration;
 import org.infinispan.configuration.parsing.ConfigurationBuilderHolder;
 import org.infinispan.configuration.parsing.Parser;
@@ -30,7 +34,6 @@ final class SubstituteGlobalComponentRegistry {
 
 @TargetClass(Parser.class)
 final class SubstituteParser {
-
     @Substitute
     private void parseJmx(XMLExtendedStreamReader reader, ConfigurationBuilderHolder holder) {
         // Ignore JMX configuration - but we need to skip to next element
@@ -43,10 +46,66 @@ final class SubstituteParser {
     }
 }
 
+@TargetClass(GlobalConfiguration.class)
+final class SubstituteGlobalConfiguration {
+    @Substitute
+    public boolean statistics() {
+        return false;
+    }
+}
+
 @TargetClass(GlobalJmxStatisticsConfiguration.class)
 final class SubstituteGlobalJmxStatisticsConfiguration {
     @Substitute
     public boolean enabled() {
         return false;
+    }
+}
+
+@TargetClass(className = "org.infinispan.jmx.AbstractJmxRegistration")
+final class SubstituteAbstractJmxRegistration {
+    @Substitute
+    public void start() {
+        // Do nothing
+    }
+
+    @Substitute
+    public void stop() {
+        // Do nothing
+    }
+
+    @Substitute
+    public final String getDomain() {
+        return "";
+    }
+
+    @Substitute
+    public final String getGroupName() {
+        return "";
+    }
+
+    @Substitute
+    public final MBeanServer getMBeanServer() {
+        return null;
+    }
+
+    @Substitute
+    public ObjectName registerExternalMBean(Object managedComponent, String groupName) throws Exception {
+        return null;
+    }
+
+    @Substitute
+    public void registerMBean(Object managedComponent) throws Exception {
+        // Do nothing
+    }
+
+    @Substitute
+    public void registerMBean(Object managedComponent, String groupName) throws Exception {
+        // Do nothing
+    }
+
+    @Substitute
+    public void unregisterMBean(ObjectName objectName) throws Exception {
+        // Do nothing
     }
 }

--- a/extensions/infinispan-embedded/runtime/src/main/java/io/quarkus/infinispan/embedded/runtime/graal/SubstituteJGroups.java
+++ b/extensions/infinispan-embedded/runtime/src/main/java/io/quarkus/infinispan/embedded/runtime/graal/SubstituteJGroups.java
@@ -2,7 +2,9 @@ package io.quarkus.infinispan.embedded.runtime.graal;
 
 import java.security.SecureRandom;
 
+import org.jgroups.JChannel;
 import org.jgroups.util.UUID;
+import org.jgroups.util.Util;
 
 import com.oracle.svm.core.annotate.Alias;
 import com.oracle.svm.core.annotate.RecomputeFieldValue;
@@ -45,5 +47,14 @@ final class SubstituteDiscardProtocol {
         // should call super.stop() but the "super" Protocol.stop() does nothing,
         // so this empty impl is OK
 
+    }
+}
+
+@TargetClass(Util.class)
+final class SubstituteJgroupsUtil {
+
+    @Substitute
+    public static void registerChannel(JChannel channel, String name) {
+        // do nothing
     }
 }

--- a/integration-tests/infinispan-client/pom.xml
+++ b/integration-tests/infinispan-client/pom.xml
@@ -80,6 +80,11 @@
             <type>test-jar</type>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.infinispan</groupId>
+            <artifactId>infinispan-component-annotations</artifactId>
+            <scope>test</scope>
+        </dependency>
 
         <!-- We override these to satisfy Infinispan server -->
         <dependency>

--- a/integration-tests/infinispan-embedded/src/main/resources/10.1.xml
+++ b/integration-tests/infinispan-embedded/src/main/resources/10.1.xml
@@ -1,9 +1,9 @@
 <infinispan
       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-      xsi:schemaLocation="urn:infinispan:config:10.0 http://www.infinispan.org/schemas/infinispan-config-10.0.xsd
+      xsi:schemaLocation="urn:infinispan:config:10.1 http://www.infinispan.org/schemas/infinispan-config-10.1.xsd
                           urn:org:jgroups http://www.jgroups.org/schema/jgroups-4.0.xsd"
-      xmlns="urn:infinispan:config:10.0"
-      xmlns:ispn="urn:infinispan:config:10.0">
+      xmlns="urn:infinispan:config:10.1"
+      xmlns:ispn="urn:infinispan:config:10.1">
    <jgroups transport="org.infinispan.remoting.transport.jgroups.JGroupsTransport">
       <!-- Load external JGroups stacks -->
       <stack-file name="udp-test" path="stacks/udp.xml"/>
@@ -85,7 +85,7 @@
    <threads>
       <thread-factory name="infinispan-factory" group-name="infinispan" thread-name-pattern="%G %i" priority="5"/>
       <!-- listener-executor -->
-      <blocking-bounded-queue-thread-pool name="infinispan-async" thread-factory="infinispan-factory"
+      <blocking-bounded-queue-thread-pool name="async-infinispan" thread-factory="infinispan-factory"
             core-threads="5" max-threads="5" queue-length="0" keepalive-time="0"/>
       <!-- listener-executor -->
       <blocking-bounded-queue-thread-pool name="infinispan-listener" thread-factory="infinispan-factory"
@@ -102,7 +102,7 @@
       <scheduled-thread-pool name="infinispan-expiration" thread-factory="infinispan-factory" />
    </threads>
 
-   <cache-container name="maximal" aliases="alias1 alias2" default-cache="local" async-executor="infinispan-async" expiration-executor="infinispan-expiration"
+   <cache-container name="maximal" aliases="alias1 alias2" default-cache="local" async-executor="async-infinispan" expiration-executor="infinispan-expiration"
                     jndi-name="java:global/infinispan/maximal"  state-transfer-executor="infinispan-state-transfer" listener-executor="infinispan-listener"
                     persistence-executor="infinispan-cached" module="org.infinispan" statistics="true" shutdown-hook="DONT_REGISTER" zero-capacity-node="false">
        <transport cluster="maximal-cluster" executor="infinispan-transport" remote-command-executor="infinispan-cached" lock-timeout="120000" stack="tcpping" node-name="a-node" machine="a" rack="b" site="c"

--- a/integration-tests/infinispan-embedded/src/main/resources/dist.xml
+++ b/integration-tests/infinispan-embedded/src/main/resources/dist.xml
@@ -1,9 +1,9 @@
 <infinispan
       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-      xsi:schemaLocation="urn:infinispan:config:10.0 http://www.infinispan.org/schemas/infinispan-config-10.0.xsd
+      xsi:schemaLocation="urn:infinispan:config:10.1 http://www.infinispan.org/schemas/infinispan-config-10.1.xsd
                           urn:org:jgroups http://www.jgroups.org/schema/jgroups-4.0.xsd"
-      xmlns="urn:infinispan:config:10.0"
-      xmlns:ispn="urn:infinispan:config:10.0">
+      xmlns="urn:infinispan:config:10.1"
+      xmlns:ispn="urn:infinispan:config:10.1">
     <!-- This is a sample way of setting up a distributed cache in a cluster with native. Note that UDP is not yet
          supported, so we have to use something like TCPPING for discovery -->
    <jgroups>

--- a/integration-tests/infinispan-embedded/src/test/java/io/quarkus/it/infinispan/embedded/InfinispanEmbeddedTestResource.java
+++ b/integration-tests/infinispan-embedded/src/test/java/io/quarkus/it/infinispan/embedded/InfinispanEmbeddedTestResource.java
@@ -14,8 +14,8 @@ import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
 public class InfinispanEmbeddedTestResource implements QuarkusTestResourceLifecycleManager {
     @Override
     public Map<String, String> start() {
-        // Since 10.0.xml is not in the classpath for native and not in the image - we have to point to the src one
-        String xmlLocation = Paths.get(System.getProperty("basedir"), "src", "main", "resources", "10.0.xml").toString();
+        // Since 10.1.xml is not in the classpath for native and not in the image - we have to point to the src one
+        String xmlLocation = Paths.get(System.getProperty("basedir"), "src", "main", "resources", "10.1.xml").toString();
         return Collections.singletonMap("quarkus.infinispan-embedded.xml-config", xmlLocation);
     }
 

--- a/test-framework/junit5/src/main/java/io/quarkus/test/junit/NativeTestExtension.java
+++ b/test-framework/junit5/src/main/java/io/quarkus/test/junit/NativeTestExtension.java
@@ -70,7 +70,7 @@ public class NativeTestExtension
             } catch (Exception e) {
 
                 failedBoot = true;
-                throw new JUnitException("Quarkus native image start failed, original cause: " + e);
+                throw new JUnitException("Quarkus native image start failed, original cause: " + e, e);
             }
         }
     }

